src/cromshell/counts/command.py

```diff
@@ -0,0 +1,288 @@
+        # If task has a key called 'subworkflowMetadata' in its first (zero) element
+        # (shard) and expand_sub_workflow parameter is set to true then rerun this
+        # function on that subworkflow
+        if "subWorkflowMetadata" in calls_metadata[call][0] and expand_sub_workflows:
+            sub_workflow_name = call
+            sub_calls = calls_metadata[sub_workflow_name]
+            print(f"{indent}SubWorkflow {sub_workflow_name}")
+
+            # For each call in subworkflow calls get the subworkflow metadata.
+            # This loop will go through each shard if task is scattered
+            for sub_call in sub_calls:
+                print_workflow_status(
+                    workflow_metadata=sub_call["subWorkflowMetadata"],
+                    indent=indent + "\t",
+                    expand_sub_workflows=expand_sub_workflows,
+                )
+
+        # If no subworkflow is found then print status summary for task
+        else:
+            print_call_status(
+                call=call, indent=indent, workflow_calls_metadata=calls_metadata
+            )
+
+
+def print_call_status(call: str, indent: str, workflow_calls_metadata: dict) -> None:
+    """
+    Prints the task name and status count
+
+    :param call: Name of the call to print
+    :param indent: Indent string given as "\t", used to indent print out
+    :param workflow_calls_metadata: The 'calls' of the workflow to process
+    :return:
+    """
+
+    shards = workflow_calls_metadata[call]
+
+    shard_status_count = get_shard_status_count(shards)
+
+    shards_done = shard_status_count.get(TaskStatuses.Done.value, 0)
+    shards_running = shard_status_count.get(TaskStatuses.Running.value, 0)
+    shards_failed = shard_status_count.get(TaskStatuses.Failed.value, 0)
+    shards_retried = shard_status_count.get(TaskStatuses.RetryableFailure.value, 0)
+    shards_unknown, unknown_shard_status = get_unknown_status(
+        shard_status_count=shard_status_count, known_statuses=TaskStatuses.list()
+    )
+
+    # Determine what color to print task summary
+    if shards_failed == 0 and shards_running == 0:
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_SUCCEEDED
+    elif shards_failed > 0 and shards_running > 0:  # Running but will fail
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_FAILING
+    elif shards_running > 0:
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_RUNNING
+    elif shards_failed > 0:
+        task_status_font = io_utils.TextStatusesColor.TASK_COLOR_FAILED
+    else:  # catch all
+        task_status_font = "yellow"
+
+    # Format and print task summary
+    formatted_task_summary = (
+        f"{indent}{call}\t{shards_running} Running, "
+        f"{shards_done} Done, {shards_retried} Preempted, {shards_failed} Failed"
+    )
+    # If unknown status present append its count to print out.
+    if shards_unknown > 0:
```

Comment:
I'd vote for this to be moved to after the failed shards are printed.

Reply:
moved to after next if statement